programs/ssl/ssl_server2.c

```diff
@@ -823,18 +823,23 @@ int sni_callback( void *p_info, mbedtls_ssl_context *ssl,
 {
     const sni_entry *cur = (const sni_entry *) p_info;
 
+    /* preserve behavior which checks for SNI match in sni_callback() for
+     * the benefits of tests using sni_callback(), even though the actual
+     * certificate assignment has moved to certificate selection callback
+     * in this application.  This exercises sni_callback and cert_callback
+     * even though real applications might choose to do this differently.
+     * Application might choose to save name and name_len in user_data for
```

Comment:
Well, they don't really need to if we keep `mbedtls_ssl_get_hs_sni()`, right?

Reply:
If an application developer new to the library were reading ssl_server2.c to see how to do things, this comment documents that splitting the work here to select cert from `p_info` is for test purposes, and could all be done in the server certificate selection callback.  Please let me know if you would like me to change or to remove the comment.  I am leaving it as-is for the moment.